### ISSUE TITLE
Access to the path is denied

### ISSUE THREAD

**Issue by grant007** [JimBobSquarePants/ImageProcessor#793]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am running the latest version of ImageProcessor
- [ ] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [X] I have searched [open](https://github.com/JimBobSquarePants/ImageProcessor/issues) and [closed](https://github.com/JimBobSquarePants/ImageProcessor/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
My .NET application is crashing many times per day. Errors similar to the one shown below are logged in the Windows Event Log - Application.

I have done everything to my knowledge to isolate the root of this error. I have set permissions on the `\App_Data\cache` to full control, but that did not remedy. I do not know what to do and would value any guidance.

```
An unhandled exception occurred and the process was terminated.
Application ID: /LM/W3SVC/5/ROOT
Process ID: 7180
Exception: System.AggregateException
Message: One or more errors occurred.
StackTrace:    at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at ImageProcessor.Web.Caching.ImageCacheBase.CacheTrimmer.TimerRelease(Func`2 trimmer)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.TimerQueueTimer.CallCallback()
   at System.Threading.TimerQueueTimer.Fire()
   at System.Threading.TimerQueue.FireNextTimers()
InnerException: System.UnauthorizedAccessException
Message: Access to the path '<redacted>\App_Data\cache\b\a\a\8\c' is denied.
StackTrace:    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileSystemEnumerableIterator`1.CommonInit()
   at System.IO.FileSystemEnumerableIterator`1..ctor(String path, String originalUserPath, String searchPattern, SearchOption searchOption, SearchResultHandler`1 resultHandler, Boolean checkHost)
   at System.IO.Directory.EnumerateFiles(String path)
   at ImageProcessor.Web.Caching.DiskCache.<TrimCacheAsync>b__12_0(CancellationToken token)
   at ImageProcessor.Web.Caching.ImageCacheBase.CacheTrimmer.<PerformTrim>d__9.MoveNext()
```

### Steps to Reproduce
I cannot reproduce the issue in development environment, but it is happening on production server.

### System Configuration
cache.config file:
```xml
<?xml version="1.0" encoding="utf-8"?>
<caching currentCache="DiskCache">
  <caches>
    <cache name="DiskCache" type="ImageProcessor.Web.Caching.DiskCache, ImageProcessor.Web" maxDays="0" memoryMaxMinutes="0" browserMaxDays="0">
      <settings>
        <setting key="VirtualCachePath" value="~/app_data/cache" />
      </settings>
    </cache>
  </caches>
</caching>
```
- ImageProcessor version: 2.8.0
- ImageProcessor.Web version (if applicable) 4.11.0
- Other ImageProcessor packages and versions:
  - ImageProcessor.Web.Config 2.6.0
- Environment (Operating system, version and so on): Windows Server 2012 R2 Datacenter
- .NET Framework version: 4.6

**Comment by ronaldbarendse** [comment]
After reformatting your comment, I see you've already provided the contents of the `cache.config` file :+1:

Setting all cache durations to 0 makes no sense (you can't and absolutely don't want to disable caching)!

This also causes the error: after the image is processed and cached, it wants to trim/delete it right away (because of the `maxDays="0"`), but it's still in use, as the client is probably still downloading it...

Simply put: your configuration is invalid and unsupported. Check the documentation for the default/correct values: https://imageprocessor.org/imageprocessor-web/configuration/#cacheconfig.

**Comment by ronaldbarendse** [comment]
BTW your application is only crashing on the production environment because that has Rapid-Fail Protection enabled: https://docs.microsoft.com/en-us/iis/configuration/system.applicationhost/applicationpools/add/failure.
